src/firefly/js/charts/ui/options/BasicOptions.jsx

```diff
@@ -86,13 +93,18 @@ export class BasicOptions extends SimpleComponent {
         const tbl_id = get(tablesource, 'tbl_id');
         const type = get(data, `${activeTrace}.type`, 'scatter');
         const noColor = !hasMarkerColor(type);
-        const noXY = hasNoXY(type);
+        const noXY = hasNoXY(type, tablesource);
+        const xColType = noXY ? '' : getColumnType(getTblById(tbl_id), get(tablesource, ['mappings', 'x'], ''));
+        const yColType = noXY ? '' : getColumnType(getTblById(tbl_id), get(tablesource, ['mappings', 'y'], ''));
+        const xNoLog = type.includes('histogram') ? true : undefined;          // histogram2d or histogram2dcontour
```

Comment:
for some cases if the log is checked, the plotly rendering run into some problems. 
- if the column is 'string' type 
- histogram2d, histrogram2dcontour. 

so the chart type and table column type for x & y are considered to be checked, but not sure if this check is needed in case no XY options is shown in the side bar for unknown type chart.